### ISSUE TITLE
Support play/pause on media already playing

### ISSUE THREAD

**Issue by daaru00** [thibauts/node-castv2-client#93]
Hello,

is it possible to support `play` and `pause` command on current playing media without launching a new video?

Something like this:
```js
client.connect(host, function() {
    // some method to load current player
    client.loadCurrentPlayer(function(err, player) {
       player.pause()
       // ...
       player.play()
    })
})
```
or directly operating on client object (like stop/volume/mute/unmute action):
```js
client.connect(host, function() {
    client.pause()
    // ...
    client.play()
})
```

**Comment by EricHier** [comment]
Any updates on this? Would be awesome to use! :)

**Comment by pedromsilvapt** [comment]
Hi. AFAIK it is possible, I've used a version of that in one of my projects. The code should be something like this, if I remember correctly.

```ts
import { Client, DefaultMediaReceiver } from 'castv2-client';

const IP_ADDR = ...;

const client = new Client();

client.on( 'status', status => {
    if ( status && status.applications instanceof Array ) {
        const app = status.applications.find( app => app.appId === DefaultMediaReceiver.APP_ID );

        if ( app != null ) {
            client.join( app, DefaultMediaReceiver, ( err, player ) => {

            } );
        } else {
            client.launch( DefaultMediaReceiver, ( err, player ) => {

            } );
        }
    }
} );

client.connect( IP_ADDR );
```

I wrote this code off the top of my head, so there might be dragons (more like bugs 😄) in there. Let me know if it works for you.

**Comment by EricHier** [comment]
Thank you @pedromsilvapt - I've tried that code. :) When I run that code, the Spotify casting stops and the Default Media Reciever takes over. The code exits with a `TypeError: Cannot read property 'mediaSessionId' of null`. Do you have an idea of what could be a solution?

**Comment by pedromsilvapt** [comment]
Hi @EricHier. Yeah, sorry, I assumed you were using the default application, which is why in the find method, I was looking for a session like this `app.appId === DefaultMediaReceiver.APP_ID`. You will have to look for the ID of the Chromecast Spotify app (you can do it by making a `console.log( status )` while the spotify app is open.

One caveat is, I'm not sure the `DefaultMediaReceiver` can control the spotify app, never tried it. Maybe the commands are the same (play, pause, etc...).

**Comment by EricHier** [comment]
Ah okay, I see. The sessions I get with `getSessions` looks like this: 
```JS
[{ appId: 'CC32E753',
    displayName: 'Spotify',
    iconUrl:
     'https://lh3.googleusercontent.com/HOX9yqNu6y87Chb1lHYqhKVTQW43oFAFFe2ojx94yCLh0yMzgygTrM0RweAexApRWqq6UahgrWYimVgK',
    isIdleScreen: false,
    launchedFromCloud: false,
    namespaces: [ [Object], [Object], [Object], [Object], [Object] ],
    sessionId: 'e91fc450-998e-4d1c-b3b6-9cc95c578654',
    statusText: 'Casting: I Want To Break Free - Single Remix',
    transportId: 'e91fc450-998e-4d1c-b3b6-9cc95c578654' }]
```

Afterwards I tried to run 
```JS
client.join(sessions[0], DefaultMediaReceiver, (err, player) => {
    player.pause();
});
```
and get the `TypeError: Cannot read property 'mediaSessionId' of null`.

Does this mean that the protocol is not the same? Or do I miss something?

**Comment by pedromsilvapt** [comment]
Yes I suppose. That particular error seems in my view to be triggered by the status reponse message, which does not contain the mediaSessionId that the DefaultMediaReceiver needs to call the pause method. I've never worked with the Spotify App, so I can't be of much help with that. Sorry 😢

**Comment by EricHier** [comment]
Okay. :) I highly appreciate your help - so awesome! ^^ I am not that experienced with this module and you helped me to get a realistic evaluation of the issue. :)
